File: src/main/java/pl/dolien/shop/purchase/Purchase.java
package pl.dolien.shop.purchase;

import lombok.*;
import pl.dolien.shop.order.Address;
import pl.dolien.shop.order.Customer;
import pl.dolien.shop.order.Order;
import pl.dolien.shop.order.OrderItem;

import java.util.Set;

@Getter
@Setter
@Builder
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;

}

File: shop-ui/src/app/common/purchase/purchase.ts
import { Address } from '../address/address';
import { Customer } from '../customer/customer';
import { OrderItem } from '../order-item/order-item';
import { Order } from '../order/order';

export class Purchase {
  customer?: Customer;
  shippingAddress?: Address;
  billingAddress?: Address;
  order?: Order;
  orderItems?: OrderItem[];
}
